### PR TITLE
IBAN for contacts

### PR DESCRIPTION
**Pull request by clenrock**
- Relevant Issues: - #241 
- Related Pull Requests: -

* * *

## What

This Pull Request adds to the repository a new field in Contact, and the ways to see them and change them in the ContactActivity.

## Why

This Pull Request is needed because the IBAN number is needed for integration.

## How

This feature can be viewed/tested within the project by going to the ContactActivity and the ChangeContactActivity.

## Alternative implementation

Other implementations that I've have considered are ...

## Notes

None

### CODE REVIEW

File: app/src/main/java/com/nervousfish/nervousfish/data_objects/Contact.java [hunk 1]
```diff
@@ -11,15 +11,21 @@
...
+@SuppressWarnings({"PMD.ImmutableField", "PMD.UselessParentheses"})
+//1. We don't want the field iban to be final.
+//2. The parentheses on line 125 are not useless.
 public final class Contact implements Serializable {
 
     private static final long serialVersionUID = -4715364587956219157L;
     private final String name;
     private final List<IKey> keys = new ArrayList<>();
+    @android.support.annotation.Nullable
+    private String iban;
```
Comment (jverbraeken) [inline app/src/main/java/com/nervousfish/nervousfish/data_objects/Contact.java]: Could be final I think. Then you can also remove the @SuppressWarning

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -24,6 +24,7 @@
     <string name="submit">Submit</string>
     <string name="type">Type</string>
     <string name="yes">Yes</string>
+    <string name="dash">-</string>
```
Comment (jverbraeken) [inline app/src/main/res/values/strings.xml]: `name="dash" translatable="false"`
Then you can remove dash from strings-nl.xml

File: app/src/main/res/values/colors.xml [hunk 3]
```diff
@@ -6,8 +6,9 @@
 
     <color name="transparent_white">#33FFFFFF</color>
     <color name="white">#ffffff</color>
-    <color name="red_fail">#dd6b55</color>
+    <color name="dark_grey">#323232</color>
 
+    <color name="red_fail">#dd6b55</color>
```
Comment (TheBonheurs) [inline app/src/main/res/values/colors.xml]: Is there any particular reason for this change up or just random?

File: app/src/main/res/layout/activity_change_contact.xml [hunk 4]
```diff
@@ -67,8 +68,8 @@
             <RelativeLayout
                 android:layout_width="match_parent"
                 android:layout_height="wrap_content"
-                app:layout_scrollFlags="scroll|enterAlways"
-                android:background="@drawable/seabackflipped">
+                android:background="@drawable/seabackflipped"
```
Comment (TheBonheurs) [inline app/src/main/res/layout/activity_change_contact.xml]: cool name 🙊

File: app/src/main/java/com/nervousfish/nervousfish/data_objects/Contact.java [hunk 5]
```diff
@@ -45,10 +65,30 @@ public Contact(final String name, final Collection<IKey> keys) {
         }
     }
 
+    /**
+     * Constructor for the {@link Contact} POJO for multiple {@link IKey}s, and
+     * the IBAN.
```
Comment (TheBonheurs) [inline app/src/main/java/com/nervousfish/nervousfish/data_objects/Contact.java]: Here you say the IBAN but @ line 43 you say an IBAN (nitpicking)

File: app/src/main/java/com/nervousfish/nervousfish/data_objects/Contact.java [hunk 6]
```diff
@@ -11,15 +11,21 @@
...
 import java.util.Collection;
 import java.util.List;
+import java.util.Objects;
 
 /**
  * A Java object representation (POJO) of a contact.
  */
+@SuppressWarnings({"PMD.ImmutableField", "PMD.UselessParentheses"})
+//1. We don't want the field iban to be final.
+//2. The parentheses on line 125 are not useless.
```
Comment (TheBonheurs) [inline app/src/main/java/com/nervousfish/nervousfish/data_objects/Contact.java]: You sure?

File: app/src/androidTest/assets/features/change_contact.feature [hunk 7]
```diff
@@ -34,6 +34,33 @@ Feature: Change contact activity
...
+    When I select the IBAN field
+    And I type <newiban> as new IBAN
+    And I press the save contact changes button
+    Then I should go to the activity I visited before the change contact activity
+
+    Examples:
+      | newiban             |
+      | NL31ABNA25285691    |
+      | Jack Sparrow        |
+      | Salazar             |
```
Comment (TheBonheurs) [inline app/src/androidTest/assets/features/change_contact.feature]: Loving the names! #underrated

File: app/src/main/java/com/nervousfish/nervousfish/data_objects/Contact.java (hunk 1 above)
Comment (clenrock) [inline app/src/main/java/com/nervousfish/nervousfish/data_objects/Contact.java]: We discussed this over Whatsapp, and we want the IBAN to be optional. That is why the attribute can't be final. The annotation can be removed though.

File: app/src/main/res/values/colors.xml (hunk 3 above)
Comment (clenrock) [inline app/src/main/res/values/colors.xml]: I found that it fitted more at the colors below that

File: app/src/main/java/com/nervousfish/nervousfish/data_objects/Contact.java (hunk 6 above)
Comment (clenrock) [inline app/src/main/java/com/nervousfish/nervousfish/data_objects/Contact.java]: Yep

**Comment by clenrock** [comment]
Closed because we don't use a string for IBAN anymore, but the IBAN object from the Bankver library

**Pull request closed without merging**